Update contributing docs

waynew opened the pull request:
### What does this PR do?

Updates the contributing docs to be in-line with the `master` branch strategy, as well as reference `pre-commit`

### What issues does this PR fix or reference?

A couple of SEPs, comments in Slack, and other places.

### Tests written?

Doc changes only

### Commits signed with GPG?

Yes

---

Note: this PR is waiting for https://github.com/saltstack/salt/pull/55765 to be merged, so contributors can really use black.

waynew committed:
Update contributing docs

We've changed to a `master` branching strategy, and we'll be working
with `black` and `pre-commit`. So there's that.

s0undt3ch commented on doc/topics/development/contributing.rst:
```diff
@@ -209,8 +247,8 @@ Fork a Repo Guide_>`_ and is well worth reading.
-    take 30 minutes.  Depending on your GitHub notification settings you may
-    also receive an email message about the test results.
+    take a couple of hours.  Depending on your GitHub notification settings
+    you may also receive an email message about the test results.
```
We send out emails through GH? 
I don't think we do, do we?

waynew replied:
GH might send emails - we don't.

waynew replied:
(they [can] send a bunch of different notifications depending on what your settings are)

waynew replied:
(they [can] send a bunch of different notifications depending on what your settings are)

waynew commented:
TODO: add a note about copyright/license headers (not needed/allowed)

waynew committed:
Note post-Neon for Black

waynew committed:
Added note about copyright headers

IIUC they're unnecessary, as salt is already under the Apache license,
and any contributions would fall under the same, or not be allowed as
contributions.
